grails-app/controllers/org/broadinstitute/orsp/StatusEventController.groovy

```diff
@@ -27,44 +30,28 @@ class StatusEventController extends AuthenticatedController {
             [issue.projectKey, period]
         }
     }
-
-    def index() {
-        QueryOptions options = new QueryOptions()
-        DateFormat format = DateFormat.getDateInstance(DateFormat.SHORT)
-        if (params.after) {
-            try {
-                options.after = format.parse(params.after)
-            } catch (ParseException e) {
-                log.error("Parse Exception: " + e)
-                log.error("Unable to parse 'after' date: " + params.after)
-                flash.error = e.getMessage()
-            }
-        }
-        if (params.before) {
-            try {
-                options.before = format.parse(params.before)
-            } catch (ParseException e) {
-                log.error("Parse Exception: " + e)
-                log.error("Unable to parse 'before' date: " + params.before)
-                flash.error = e.getMessage()
-            }
-        }
-        if (params.project_type) {
-            options.issueTypeNames = [IssueType.valueOfController(params.project_type).name]
-        }
-        Map<IssueType, Collection<Issue>> groupedIssues = getGroupedIssues(options)
-        Collection<Issue> irbs = groupedIssues.getOrDefault(IssueType.IRB, Collections.emptyList())
-        Collection<Issue> nonIrbs = groupedIssues.getOrDefault(IssueType.NE, Collections.emptyList()) +
-                groupedIssues.getOrDefault(IssueType.NHSR, Collections.emptyList())
-        Map<String, Period> periodMap = calculateIssuePeriods(irbs + nonIrbs)
-        [irbs: irbs, nonIrbs: nonIrbs, periodMap: periodMap]
-    }
-
+    // TODO: unused, this must be re-implemented
     def project() {
         Issue issue = queryService.findByKey(params.id)
         // Sort ascending and create DTOs
-        List<StatusEventService.StatusEventDTO> eventDTOs = statusEventService.getStatusEventDTOs(params.id)
+        List<StatusEventDTO> eventDTOs = statusEventService.getStatusEventDTOs(params.id)
         [statusEvents: eventDTOs, issue: issue]
     }
 
+    def findQaEventReport() {
+        UtilityClass.registerQaReportIssueMarshaller()
+        Collection<String> issueTypeNames = new ArrayList<String>()
+
+        if (params.tab) {
+            issueTypeNames = params.tab == NO_IRB ? [IssueType.NE.name, IssueType.NHSR.name] : [IssueType.IRB.name]
+        }
```

Comment:
Another way to avoid mutation is to instantiate the collection in the conditional. Something like (I used EnumSet just for flavor):
```
        Collection<String> issueTypeNames = (params?.tab == NO_IRB) ? 
                EnumSet.of(IssueType.NE, IssueType.NHSR)*.getName() : 
                EnumSet.of(IssueType.IRB)*.getName() 
```

Reply:
Cool, taking note. Thanks for the recommendation.